### ISSUE TITLE
Bug: processing of DEVELOPER_CONFIGS

### ISSUE THREAD

**Issue by torvista** [zencart/zencart#3623]
Currently
`foreach (DEVELOPER_CONFIGS as $key) {
if (defined($key)) {`

As it is, the $key variable actually holds the value of the array element, not the key. So the check for the existence of that key as a constant, fails: none of the array values are used.

`foreach (DEVELOPER_CONFIGS as $key =>$value) {`
This correctly sets $key as the key/constant name.

However, subsequently, I am not too clear if this is correct either:

```
$value = constant($key);
                    if (null === $value) continue;
                    $sql = "update " . $this->dbPrefix . "configuration set configuration_value = '" . $this->db->prepareInput($value) . "' where configuration_key = '" . $this->db->prepareInput($key) . "'";
```

In addition to this being fixed, something that would complete this developer help for overrides, would be an else clause for undefined constants: to add them (and create a log record of the additions).

**Comment by drbyte** [comment]
Bug: yes. Fixed. Originally I'd written it as actual defines, but then I knew that people would want it to scale so I changed it to an array. Evidently I still had my defines in place when testing so I didn't catch this. Thanks.

> In addition to this being fixed, something that would complete this developer help for overrides, would be an else clause for undefined constants: to add them (and create a log record of the additions).

I really don't understand what you're saying here. What undefined constants? What purpose would a log serve here? This is for initial dev-mode setup; what's the logging for?
